### ISSUE TITLE
Colors in diff highlighting incorrect when deleted line starts with dash-dash

### ISSUE THREAD

**Issue by lurkie** [jonas/tig#1102]
Given an SQL file, where some comment lines are in it and these comment lines start with a double dash: `--`

When deleting such a comment line, tig displays it with a triple dash: `---`, which is correct in the context of a diff. But the assigned color for that line is not red as one would expect for a deleted line, but yellow like AuthorDate and CommitDate, and the start of a diff for a file (`diff --git a/file.sql b/file.sql\n index ...\n--- a/file.sql\n+++ b/file.sql`, although the 'index' line is not yellow)

The same issue appears as well when adding a line starting with a double plus: `++`

Using version: 2.3.0, Ubuntu 18.04

**Comment by krobelus** [comment]
How to reproduce? If I do something like

```shell
$ printf -- '--\n' > new
$ git diff --no-index new /dev/null | tig
```

the `---` line is red.

**Comment by koutcher** [comment]
See #1099, the issue with `---` and `+++` was fixed in 2.5.0.
